pkg/storer/reserve.go

```diff
@@ -225,14 +289,32 @@ func (db *DB) evictBatch(ctx context.Context, batchID []byte, upToBin uint8) (er
 
 		var evicted int
 
-		err = db.reserve.IterateBatchBin(ctx, db.repo, b, batchID, func(address swarm.Address) (bool, error) {
-			err := db.removeChunk(ctx, address, batchID)
+		type evictItems struct {
+			address swarm.Address
+			batchID []byte
+		}
+
+		itemsToEvict := make([]evictItems, 0)
```

Comment:
The `0` is unnecessary. If you use `var itemsToEvict []evictItems`, you avoid allocation in the case when there is no item to evict.